core/src/main/kotlin/org/evomaster/core/search/gene/collection/FixedMapGene.kt

```diff
@@ -0,0 +1,89 @@
+package org.evomaster.core.search.gene.collection
+
+import org.evomaster.core.logging.LoggingUtil
+import org.evomaster.core.search.gene.*
+import org.evomaster.core.search.service.Randomness
+import org.evomaster.core.search.service.mutator.MutationWeightControl
+import org.evomaster.core.search.service.mutator.genemutation.AdditionalGeneMutationInfo
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * FIXME: this needs to be refactored, as at the moment the
+ * keys are strings that are fixed.
+ * Keys should be of any basic type, and should be modifiable.
+ *
```

Comment:
what does it mean here to be `fixed`?  i m a bit confused... i don't see here anything constraining that keys should be `StringGene`, nor that are immutable (eg with `CustomMutationRateGene`)

Reply:
this comment is quite old, and not updated to the latest implementation. now new doc is added.